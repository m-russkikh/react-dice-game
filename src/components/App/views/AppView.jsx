import React from 'react';
import { Container } from 'reactstrap';
import 'bootstrap/dist/css/bootstrap.css';

import Balance from '../../Balance';
import BetAmount from '../../BetAmount';
import BetNumber from '../../BetNumber';
import ButtonsPanel from '../../ButtonsPanel';
import ProbablyFairHash from '../../ProbablyFairHash';
import ResultArea from '../../ResultArea';
import NumberOfBets from '../../NumberOfBets';
import Table from '../../Table';

function AppView() {
  return (
    <Container>
      <Balance />
      <BetAmount />
      <BetNumber />
      <NumberOfBets />
      <ButtonsPanel />
      <ProbablyFairHash />
      <ResultArea />
      <Table />
    </Container>
  );
}

export default AppView;
