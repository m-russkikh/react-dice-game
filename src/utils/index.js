import { MAX_BET_NUMBER } from '../constants/diceGame';


export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getBetNumber() {
  return Math.floor(Math.random(0, 1) * MAX_BET_NUMBER) + 1;
}
