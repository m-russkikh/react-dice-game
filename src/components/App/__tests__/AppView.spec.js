import { getShallowWrapper } from '../../../utils/testUtils';
import AppView from '../views/AppView';


describe('AppView', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = getShallowWrapper(AppView);
  });

  describe('render', () => {
    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
