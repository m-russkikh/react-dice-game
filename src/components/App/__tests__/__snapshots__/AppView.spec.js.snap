// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView render should display correct 1`] = `
"<Container tag=\\"div\\">
  <Connect(BalanceContainer) />
  <Connect(BetAmountContainer) />
  <Connect(BetNumberContainer) />
  <Connect(NumberOfBetsContainer) />
  <ButtonsPanelView />
  <Connect(ProbablyFairHashView) />
  <Connect(ResultAreaView) />
  <Connect(TableView) />
</Container>"
`;
